sympy/core/basic.py

```diff
@@ -342,9 +342,15 @@ def __eq__(self, other):
             return True
 
         if not isinstance(other, Basic):
+            if iterable(other, exclude=(str, NotIterable)
+                    ) and not hasattr(other, '_sympy_'):
+                # XXX iterable self should have it's own __eq__
+                # method if the path gives a false negative
+                # comparison
+                return False
             try:
                 other = _sympify(other)
-            except SympifyError:
+            except (SympifyError, SyntaxError):
                 return NotImplemented
```

Comment:
Why is this catching SyntaxError? There can't be a good reason to catch that here.

Reply:
I copied this over from `Expr.__eq__`, should it be removed in both places?

Reply:
This is also in `numbers.py`. Possibly it is to catch errors with `str` parsed expressions in classes that (unfortunately) use `sympify` over `_sympify`

Reply:
```python
from sympy import *
Tuple("2x")
```
Gives a `SyntaxError`

Reply:
How exactly does calling `_sympify` lead to the `SyntaxError`?
```python
In [65]: from sympy.core.sympify import _sympify

In [66]: _sympify(('2x', 'x'))
---------------------------------------------------------------------------
SympifyError 
```

Reply:
Sorry for the noise, I had already changed `Tuple.__new__` locally to use `_sympify`. With actual master I see the `SyntaxError`:
```python
In [1]: from sympy.core.sympify import _sympify

In [2]: _sympify(('2x', 'x'))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
ValueError: Error from parse_expr with transformed code: "Integer (2 )Symbol ('x' )"

The above exception was the direct cause of the following exception:

SyntaxError  
```

Reply:
In case you're looking into solving all errors related to changing `sympify` to `_sympify` in `Tuple`, I do have a WIP branch to solve the issues related to `Qubit` (it's using `Tuple` to convert "0" and "1" strings to and I replaced it with an explicit conversion)

Reply:
Maybe this PR is helpful regarding this topic: https://github.com/sympy/sympy/pull/22643